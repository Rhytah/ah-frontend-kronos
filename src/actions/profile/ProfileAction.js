import axios from 'axios';
import { toastSuccess, toastFailure } from '../../utils/toast';
import 'react-toastify/dist/ReactToastify.css';
import history from '../../utils/history';

import { profileActionTypes } from '../actionTypes';
import URLS from '../../urlConnections';

export const fetchProfile = () => (dispatch) => {
  dispatch({
    type: profileActionTypes.VIEW_PROFILE_REQUEST,
  });
  return axios.get(`${URLS.BACKEND_URL}/api/profiles/${sessionStorage.getItem('username')}`, {
    headers: {
      'content-type': 'application/json',
      Authorization: `Bearer ${sessionStorage.getItem('token')}`,
    },
  })
    .then((response) => {
      dispatch({
        type: profileActionTypes.VIEW_PROFILE_SUCCESS,
        payload: response.data.profile,
      });
    })
    .catch((error) => {
      dispatch({
        type: profileActionTypes.VIEW_PROFILE_FAILED,
        payload: error.response.data.profile.detail,
      });
    });
};


export const editProfile = newProfile => (dispatch) => {
  dispatch({
    type: profileActionTypes.VIEW_PROFILE_REQUEST,
  });
  return axios.put(`${URLS.BACKEND_URL}/api/profiles/${sessionStorage.getItem('username')}/edit`, newProfile,
    {
      headers: {
        'content-type': 'application/json',
        Authorization: `Bearer ${sessionStorage.getItem('token')}`,
      },
    })
    .then((response) => {
      dispatch({
        type: profileActionTypes.EDIT_PROFILE_SUCCESS,
        payload: response.data.profile,
      });
      toastSuccess('Profile updated Successfully!', 'A');
      history.push('/viewProfile');
    })
    .catch((error) => {
      dispatch({
        type: profileActionTypes.EDIT_PROFILE_FAILED,
        payload: error.response.data.profile.detail,
      });
      toastFailure('Update failed, try again!', 'A');
    });
};
