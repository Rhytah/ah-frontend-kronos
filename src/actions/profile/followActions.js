import axios from 'axios';

import { followActionTypes, profileActionTypes } from '../actionTypes';

export const followAuthorAction = username => dispatch => axios
  .post(
    `https://ah-backend-kronos-staging.herokuapp.com/api/profiles/${username}/follow`,
    { body: {} },
    {
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${sessionStorage.getItem('token')}`,
      },
    },
  )
  .then((response) => {
    dispatch({
      type: followActionTypes.FOLLOW_AUTHOR_SUCCESS,
      payload: response.data,
    });
  })
  .catch((error) => {
    dispatch({
      type: followActionTypes.FOLLOW_AUTHOR_FAIL,
      payload: error.response,
    });
  });
export const unfollowAuthorAction = username => dispatch => axios
  .delete(
    `https://ah-backend-kronos-staging.herokuapp.com/api/profiles/${username}/follow`,

    {
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${sessionStorage.getItem('token')}`,
      },
    },
  )
  .then((response) => {
    dispatch({
      type: followActionTypes.UNFOLLOW_AUTHOR_SUCCESS,
      payload: response.data,
    });
  })
  .catch((error) => {
    dispatch({
      type: followActionTypes.UNFOLLOW_AUTHOR_FAILURE,
      payload: error.response,
    });
  });
export const fetchAuthorProfile = username => dispatch => axios
  .get(`https://ah-backend-kronos-staging.herokuapp.com/api/profiles/${username}`, {
    headers: {
      'content-type': 'application/json',
      Authorization: `Bearer ${sessionStorage.getItem('token')}`,
    },
  })
  .then((response) => {
    dispatch({
      type: profileActionTypes.VIEW_AUTHOR_PROFILE_SUCCESS,
      payload: response.data.profile,
    });
  })
  .catch((error) => {
    dispatch({
      type: profileActionTypes.VIEW_PROFILE_FAILED,
      payload: error.response.data.profile.detail,
    });
  });

export const fetchFollowing = username => dispatch => axios
  .get(`https://ah-backend-kronos-staging.herokuapp.com/api/profiles/${username}/followers`, {
    headers: {
      'content-type': 'application/json',
      Authorization: `Bearer ${sessionStorage.getItem('token')}`,
    },
  })
  .then((response) => {
    dispatch({
      type: profileActionTypes.FETCH_FOLLOWING_PROFILES,
      payload: response.data.followers,
    });
  });
