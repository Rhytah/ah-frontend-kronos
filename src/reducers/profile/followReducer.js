import { followActionTypes, profileActionTypes } from '../../actions/actionTypes';
import { initialProfileState } from '../initialState';

const followReducer = (
  state = {
    profile: initialProfileState.profile.following,
    isFollowing: false,
    authorProfiles: [],
    authorProfile: {},
    followers: [],
  },
  action,
) => {
  switch (action.type) {
    case followActionTypes.FOLLOW_AUTHOR_SUCCESS: {
      return { ...state, payload: action.payload, isFollowing: true };
    }
    case followActionTypes.UNFOLLOW_AUTHOR_SUCCESS: {
      return { ...state, payload: action.payload, isFollowing: false };
    }
    case followActionTypes.UNFOLLOW_AUTHOR_FAILURE:
      return {
        ...state,
        payload: action.payload,
        isFollowing: true,
      };
    case profileActionTypes.VIEW_AUTHOR_PROFILE_SUCCESS: {
      return { ...state, authorProfile: action.payload };
    }
    case profileActionTypes.FETCH_FOLLOWING_PROFILES:
      return {
        ...state,
        followers: action.payload,
      };
    default:
      return state;
  }
};
export default followReducer;
