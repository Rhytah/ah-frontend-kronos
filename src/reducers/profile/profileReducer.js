import { profileActionTypes } from '../../actions/actionTypes';
import { initialProfileState } from '../initialState';


export default function (state = initialProfileState, action) {
  switch (action.type) {
    case profileActionTypes.VIEW_PROFILE_REQUEST:
      return {
        ...state,
        isLoading: true,
      };
    case profileActionTypes.VIEW_PROFILE_SUCCESS:
      return {
        ...state,
        profile: action.payload,
        isLoading: false,
      };
    case profileActionTypes.VIEW_PROFILE_FAILED:
      return {
        ...state,
        error: action.payload,
        isLoading: false,
      };
    case profileActionTypes.EDIT_PROFILE_SUCCESS:
      return {
        ...state,
        profile: action.payload,
        isLoading: false,
        editSuccess: true,
      };
    case profileActionTypes.EDIT_PROFILE_FAILED:
      return {
        ...state,
        profile: action.payload,
        isLoading: false,
        editFailure: true,
      };
    default:
      return state;
  }
}
