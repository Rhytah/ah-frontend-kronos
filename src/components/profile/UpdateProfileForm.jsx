/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable no-shadow */
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import React, { Component } from 'react';

import { fetchProfile, editProfile } from '../../actions/profile/ProfileAction';
import { storage } from '../../firebase';
import Loading from '../common/Loading';

import '../../assets/scss/Profile.scss';

export class UpdateProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstName: '',
      lastName: '',
      userName: '',
      image: '',
      dateOfBirth: '',
      bio: '',
      Errorfirstname: '',
      Errorlastname: '',
      ErrordateOfBirth: '',
      Errorbio: '',
      progress: 0,
    };
  }

  componentDidMount() {
    const { fetchProfile } = this.props;
    fetchProfile();
  }

  componentWillReceiveProps(nextProps) {
    const { profile } = nextProps;
    if (profile.username) {
      this.setState({
        firstName: profile.first_name,
        lastName: profile.last_name,
        userName: profile.username,
        image: profile.image,
        dateOfBirth: profile.date_of_birth,
        bio: profile.bio,
      });
    }
  }

  handleChange = (event) => {
    event.preventDefault();
    this.setState({ [event.target.name]: event.target.value });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const {
      firstName,
      lastName,
      userName,
      image,
      dateOfBirth,
      bio,
    } = this.state;
    const ProfileData = {
      first_name: firstName,
      last_name: lastName,
      username: userName,
      image,
      date_of_birth: dateOfBirth,
      bio,
    };
    const { editProfile } = this.props;
    editProfile(ProfileData, this.props);
  }

  validateFormData = () => {
    const {
      firstName, lastName, bio,
    } = this.state;
    this.setState({
      Errorfirstname: firstName.length > 2 && firstName.length < 50 ? null : 'firstname must be atleast 3 characters but not more than 50',
      Errorlastname: lastName.length > 2 && lastName.length < 50 ? null : 'lastname must be atleast 3 characters but not more than 50',
      Errorbio: bio.length < 100 && bio.length > 12 ? null : 'Bio should not be less than 12 words or more than 100 words',
    });
  }

  uploadImage(files) {
    const fileload = storage.ref(`profile/${files[0].name}`).put(files[0]);
    fileload.on('state_changed',
      (snapshot) => {
        const progress = Math.round((snapshot.bytesTransferred / snapshot.totalBytes) * 100);
        this.setState({
          progress,
        });
      },
      (error) => {
        this.setState({
          progress: `Failed to upload file! ${error}`,
        });
      },
      () => {
        storage.ref('profile').child(files[0].name).getDownloadURL().then((url) => {
          const image = {
            image: url,
          };
          this.setState({ image: image.image });
        });
      });
  }

  render() {
    const {
      firstName, lastName, userName, dateOfBirth, bio, image,
      Errorfirstname,
      Errorlastname,
      ErrordateOfBirth,
      Errorbio,
      progress,
    } = this.state;
    const { loading } = this.props;
    return (
      <React.Fragment>
        {loading ? <Loading className="profile-loader" />
          : (
            <form className="container profile profile-view view-profile-form" id="profile">
              <div className="form-row profile-row">
                <div className="col-md-4 relative">
                  {(!image || image === 'url') ? (
                    <img
                      alt="profileimage"
                      className="avatar-bg center"
                      src="https://vignette.wikia.nocookie.net/caramella-girls/images/9/99/Blankpfp.png/revision/latest?cb=20190122015011"
                    />
                  )
                    : (<img className="avatar-bg center" name="image" alt="profileimage" src={image} />)}
                  <input
                    type="file"
                    onChange={event => this.uploadImage(event.target.files)}
                    className="upload-button mt-3"
                    id="profile"
                  />
                  <div className="progress">
                    <div
                      className="progress-bar"
                      role="progressbar"
                      style={{ width: `${progress}%` }}
                      aria-valuenow={progress}
                      aria-valuemin="0"
                      aria-valuemax="100"
                    >
                      {`${progress}%`}
                    </div>
                  </div>
                </div>
                <div className="col-md-8">
                  <h1>Profile </h1>
                  <hr />
                  <div className="form-row">
                    <div className="col-md-6 form-group">
                      <label htmlFor="firstname">
                    First Name
                        <input type="text" className="form-control" placeholder="firstName" value={firstName} name="firstName" required onChange={this.handleChange} onKeyUp={this.validateFormData} />
                      </label>
                      {Errorfirstname && (<p className="Error">{Errorfirstname}</p>)}
                    </div>
                    <div className="col-md-6 form-group">
                      <label htmlFor="lastname">
                    Last Name
                        <input type="text" className="form-control" value={lastName} name="lastName" placeholder="LastName" required onChange={this.handleChange} onKeyUp={this.validateFormData} />
                      </label>
                      {Errorlastname && (<p className="Error">{Errorlastname}</p>)}
                    </div>
                    <div className="col-md-6 form-group">
                      <label htmlFor="username">
                    User Name
                        <h5 type="text" className="form-control" name="userName" autoComplete="off">{userName}</h5>
                      </label>
                    </div>
                    <div className="col-md-6 form-group">
                      <label htmlFor="date">
                    Date Of Birth
                        <input type="date" className="form-control" value={dateOfBirth} name="dateOfBirth" autoComplete="off" onChange={this.handleChange} />
                      </label>
                      {ErrordateOfBirth && (<p className="Error">{ErrordateOfBirth}</p>)}
                    </div>
                  </div>
                  <div className="form-row col-sm-12 col-md-12">
                    <label htmlFor="bio">
                    Bio Data
                      <textarea type="text" autoComplete="off" className="form-control" rows="4" cols="80" value={bio} name="bio" placeholder="Talk about Yourself" onChange={this.handleChange} onKeyUp={this.validateFormData} />
                    </label>
                    {Errorbio && (<p className="Error">{Errorbio}</p>)}
                  </div>
                  <hr />
                  <div className="form-row col-md-12 content-right">
                    <button className="btn form-btn" type="submit" onClick={this.handleSubmit}>Update </button>
                    <a className="button-dager" href="/viewProfile" type="cancel">Cancel </a>
                  </div>
                </div>
              </div>
            </form>
          )}
      </React.Fragment>
    );
  }
}

export const mapStateToProps = ({ profile }) => ({
  profile: profile.profile,
  loading: profile.isLoading,
});

UpdateProfile.propTypes = {
  fetchProfile: PropTypes.func,
}.isRequired;

UpdateProfile.defaultProps = {
  fetchProfile: () => {},
};

export default connect(
  mapStateToProps,
  { fetchProfile, editProfile },
)(UpdateProfile);
