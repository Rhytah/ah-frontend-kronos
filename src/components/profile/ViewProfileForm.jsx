/* eslint-disable no-shadow */
import { connect } from 'react-redux';
import { ToastContainer } from 'react-toastify';
import PropTypes from 'prop-types';
import React from 'react';

import { fetchProfile } from '../../actions/profile/ProfileAction';
import Loading from '../common/Loading';

import '../../assets/scss/Profile.scss';

export class ViewProfile extends React.Component {
  componentDidMount() {
    const { viewProfile } = this.props;
    viewProfile();
  }

  render() {
    /* eslint-disable camelcase */
    const {
      profile: {
        first_name, last_name, username, date_of_birth, bio, image,
      },
    } = this.props;
    const { loading } = this.props;
    return (
      <React.Fragment>
        <ToastContainer />
        <div className="view-profile-form">
          <div className="container profile profile-view" id="profile">
            {loading ? <Loading className="profile-loader" />
              : (
                <form>
                  <div className="form-row profile-row">
                    <div className="col-md-4 relative">
                      {(!image || image === 'url') ? (
                        <img
                          alt="profileimage"
                          className="avatar-bg center"
                          src="https://vignette.wikia.nocookie.net/caramella-girls/images/9/99/Blankpfp.png/revision/latest?cb=20190122015011"
                        />
                      )
                        : (<img className="avatar-bg center" alt="profileimage" src={image} />)}
                    </div>
                    <div className="col-md-8">
                      <h1>Profile </h1>
                      <hr />
                      <div className="form-row">
                        <div className="col-md-6">
                          <div className="form-group">
                            <h5 htmlFor="firstname" className="profile-label">Firstname </h5>
                            <p type="text" name="firstname">
                              {first_name}
                            </p>
                          </div>
                        </div>
                        <div className="col-md-6">
                          <div className="form-group">
                            <h5 htmlFor="lastname" className="profile-label">Lastname </h5>
                            <p type="text" name="lastname">
                              {last_name}
                            </p>
                          </div>
                        </div>
                        <div className="col-md-6">
                          <div className="form-group">
                            <h5 htmlFor="username" className="profile-label">Username </h5>
                            <p type="text" name="username">
                              {username}
                            </p>
                          </div>
                        </div>
                        <div className="col-md-6">
                          <div className="form-group">
                            <h5 htmlFor="date" className="profile-label">Date of Birth </h5>
                            <p type="date" name="date">
                              {date_of_birth}
                            </p>
                          </div>
                        </div>
                      </div>
                      <div className="form-row">
                        <div className="col-sm-12 col-md-12">
                          <h5 htmlFor="bio" className="profile-label">Bio </h5>
                          <p type="text" name="bio">
                            {bio}
                          </p>
                        </div>
                      </div>
                      <div className="form-row" />
                      <hr />
                    </div>
                  </div>
                </form>
              )}
          </div>
        </div>
      </React.Fragment>
    );
  }
}


export const mapStateToProps = ({ profile }) => ({
  profile: profile.profile,
  loading: profile.isLoading,
});

ViewProfile.propTypes = {
  viewProfile: PropTypes.func,
}.isRequired;

ViewProfile.defaultProps = {
  viewProfile: () => {},
};

export default connect(
  mapStateToProps,
  { viewProfile: fetchProfile },
)(ViewProfile);
