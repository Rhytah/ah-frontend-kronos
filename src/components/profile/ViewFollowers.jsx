/* eslint-disable no-shadow */
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import React from 'react';

import { fetchFollowing } from '../../actions/profile/followActions';

import '../../assets/scss/Profile.scss';

export const ViewFollowers = (props) => {
  const { followers } = props;
  const followersList = followers.map(follower => (
    <div key={follower.follow}>{follower.follow}</div>
  ));
  // eslint-disable-next-line react/destructuring-assignment
  let followersDisplay;
  if (followers.length === 0) {
    followersDisplay = (
      <div>
        <span className="follow">No followers</span>
      </div>
    );
  } else {
    followersDisplay = (
      <div>
        <span className="follow">Followers</span>
        <span className="followers">{followersList}</span>
      </div>
    );
  }

  return <div>{followersDisplay}</div>;
};
export const mapStateToProps = (state) => {
  const { followers } = state.followReducer;
  return { followers };
};

ViewFollowers.propTypes = {
  fetchFollowing: PropTypes.func,
  followers: PropTypes.arrayOf(PropTypes.shape({})),
}.isRequired;

ViewFollowers.defaultProps = {
  fetchFollowing: () => {},
  followers: [],
};

export default connect(
  mapStateToProps,
  { fetchFollowing },
)(ViewFollowers);
