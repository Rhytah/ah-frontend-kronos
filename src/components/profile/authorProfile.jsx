/* eslint-disable no-shadow */
/* eslint-disable react/no-unused-state */
/* eslint-disable camelcase */
import { connect } from 'react-redux';
import { ToastContainer } from 'react-toastify';
import PropTypes from 'prop-types';
import React from 'react';
import ViewFollowers from './ViewFollowers';
import FollowButton from '../common/FollowButton';
import {
  fetchAuthorProfile,
  followAuthorAction,
  unfollowAuthorAction,
  fetchFollowing,
} from '../../actions/profile/followActions';

import Loading from '../common/Loading';

import '../../assets/scss/Profile.scss';

export class AuthorProfile extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      username1: '',
    };
  }

  componentWillMount() {
    const {
      fetchProfileAuthor,
      match: {
        params: { username },
      },
    } = this.props;
    fetchProfileAuthor(username);
  }

  unfollowUser = () => {
    const { username } = this.state;
    unfollowAuthorAction(username);
  };

  followUser = () => {
    const { username } = this.state;
    followAuthorAction(username);
  };

  fetchFollowers = (name) => {
    const { fetchFollowing } = this.props;
    fetchFollowing(name);
  };

  render() {
    const { authorProfile, loading, followers } = this.props;
    let name;
    if (!authorProfile.username) {
      name = 'user';
      return <div>nothing to display</div>;
    }
    name = '';
    if (authorProfile) {
      name = authorProfile.username;
      this.fetchFollowers(name);
    } else {
      name = '';
    }
    return (
      <React.Fragment>
        <ToastContainer />
        <div className="view-profile-form">
          <div className="container profile profile-view" id="profile">
            {name}
            {loading ? (
              <Loading className="profile-loader" />
            ) : (
              <form>
                <div className="form-row profile-row">
                  <div className="col-md-4 relative">
                    {!authorProfile.image || authorProfile.image === 'url' ? (
                      <img
                        alt="profileimage"
                        className="avatar-bg center"
                        src="https://cdn.pixabay.com/photo/2019/06/21/11/46/squirrel-4289250__480.jpg"
                      />
                    ) : (
                      <img
                        className="avatar-bg center"
                        alt="profileimage"
                        src={authorProfile.image}
                      />
                    )}
                    {sessionStorage.getItem('username') !== authorProfile.username ? (
                      <FollowButton
                        username={authorProfile.username}
                        following={authorProfile.following}
                        followUser={this.followUser}
                        unfollowUser={this.unfollowUser}
                      />
                    ) : null}
                  </div>
                  <div className="col-md-8">
                    <h1>Profile </h1>
                    <hr />
                    <div className="form-row">
                      <div className="col-md-6">
                        <div className="form-group">
                          <h5 htmlFor="firstname" className="profile-label">
                            Full Name
                          </h5>
                          <p type="text" name="firstname">
                            {authorProfile.first_name}
                            <span />
                            {authorProfile.last_name}
                          </p>
                        </div>
                      </div>

                      <div className="col-md-6">
                        <div className="form-group">
                          <h5 htmlFor="username" className="profile-label">
                            Username
                          </h5>
                          <p type="text" name="username">
                            {authorProfile.username}
                          </p>
                        </div>
                      </div>
                      <div className="col-md-6" />
                    </div>
                    <div className="form-row">
                      <div className="col-sm-12 col-md-12">
                        <h5 htmlFor="bio" className="profile-label">
                          Bio
                        </h5>
                        <p type="text" name="bio">
                          {authorProfile.bio}
                        </p>
                      </div>
                    </div>
                    <div className="form-row">
                      {' '}
                      <div className="col-md-6">
                        <ViewFollowers usernamel={name} followers={followers} />
                      </div>
                    </div>
                    <hr />
                  </div>
                </div>
              </form>
            )}
          </div>
        </div>
      </React.Fragment>
    );
  }
}

export const mapStateToProps = (state) => {
  const { authorProfile } = state.followReducer;
  return { authorProfile };
};

AuthorProfile.propTypes = {
  fetchProfileAuthor: PropTypes.func,
  username: PropTypes.string,
}.isRequired;

AuthorProfile.defaultProps = {
  fetchProfileAuthor: () => {},
  authorProfile: {},
  username: '',
};

export default connect(
  mapStateToProps,
  {
    fetchProfileAuthor: fetchAuthorProfile,
    fetchFollowing,
  },
)(AuthorProfile);
