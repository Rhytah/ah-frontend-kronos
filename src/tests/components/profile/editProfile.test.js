import React from 'react';
import { shallow } from 'enzyme';
import { UpdateProfile, mapStateToProps } from '../../../components/profile/UpdateProfileForm';

// test that UpadteProfile component
describe('<UpadteProfile />', () => {
  let wrapper;
  const props = {
    profile: {
      profile: {
        firstname: 'kronos',
        lastname: 'manuel',
        username: 'devfest',
        bio: 'Programing is my style',
        image: '',
      },
    },
    editProfile: jest.fn(),
    fetchProfile: jest.fn(),
    componentWillReceiveProps: jest.fn(),
    history: {
      push: jest.fn(),
    },
  };
  beforeEach(() => {
    wrapper = shallow(<UpdateProfile {...props} />);
  });
  it('renders without fail', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('it should map state to props', () => {
    const initialState = {
      profile: {},
    };
    expect(mapStateToProps(initialState)).toEqual({
      profile: undefined,
      loading: undefined,
    });
  });
  it('should change to profile page on edit success', () => {
    const nextProps = {
      editSuccess: true,
      profile: {
        firstname: 'kronos',
        lastname: 'manuel',
        username: 'devfest',
        bio: 'Programing is my style',
        image: '',
      },
    };
    wrapper.setProps({ ...nextProps });
    expect(props.history.length).toBe(undefined);
  });
  it('should handle change', () => {
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: 'first_name',
        value: 'kronosdevs',
      },
    };
    const wrapperInst = wrapper.instance();
    wrapperInst.handleChange(event);
    wrapperInst.handleSubmit(event);
    wrapperInst.validateFormData();
    expect(wrapperInst.state.first_name).toBe('kronosdevs');
  });
});
