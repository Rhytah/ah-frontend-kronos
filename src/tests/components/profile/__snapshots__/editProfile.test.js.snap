// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpadteProfile /> renders without fail 1`] = `ShallowWrapper {}`;
